melodies_monet/plots/surfplots.py

```diff
@@ -138,7 +157,66 @@ def map_projection(f):
         raise NotImplementedError('Projection not defined for new model. Please add to surfplots.py')
     return proj
 
-def make_spatial_bias(df, column_o=None, label_o=None, column_m=None, 
+def get_utcoffset(lat,lon):
+    """get UTC offset in hour based on a point (lat/lon)
+
+    Parameters
+    ----------
+    lat : latitude of a single point
+    lon : longitude of a single point
+        
+    Returns
+    -------
+    UTC offset in hour
+
+    """
+
+    from datetime import datetime
+    import pytz
+    from datetime import datetime, timezone
+
+    tf = TimezoneFinder()
```

Comment:
Should you be importing this within this function? Since you commented the module-level import?

Reply:
Yes, let's move this out of this function and up to the top of the code instead. Maybe we want to call this function for other uses.

Reply:
Yes, let's move this out of this function and up to the top of the code instead. Maybe we want to call this function for other uses.

Reply:
@rschwant not sure what you mean. This line is within a function. My comment was about `TimezoneFinder` not being imported anywhere, since

https://github.com/NOAA-CSL/MELODIES-MONET/blob/a873dd268c475b67c030007068c8b852da22348d/melodies_monet/plots/surfplots.py#L25

is commented

Reply:
Oh I misunderstood. I see what you mean. I'm open if we don't want to update the dependencies of MELODIES-MONET to move all of these packages to only import if you call this function. I'm thinking this will not be the default function because it will be very slow and we do not yet call it in the driver. Users can then load this package if they need it separately. Do you all agree?

Reply:
Yeah, I think the faster longitude-based method could be the default. And I think we could offload these options/details to MONET, but that can be after this PR. For now, the `from timezonefinder import TimezoneFinder` should be within the function so that only users who want reg need to install it.

Reply:
I have a function get_utcoffset in the code in case in the future we want to calculate timezone more accurately. The function get_utcoffset will use below packages:
    from datetime import datetime
    import pytz
    from datetime import datetime, timezone
    from timezonefinder import TimezoneFinder

I will put TimezoneFinder within the function as above.